modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -1508,6 +1508,21 @@ CV_EXPORTS_W void filter2D( InputArray src, OutputArray dst, int ddepth,
                             InputArray kernel, Point anchor = Point(-1,-1),
                             double delta = 0, int borderType = BORDER_DEFAULT );
 
+class CV_EXPORTS_W_PARAMS Filter2DParams
+{
+public:
+    CV_PROP_RW int anchorX = -1;
+    CV_PROP_RW int anchorY = -1;
+    CV_PROP_RW int borderType = BORDER_DEFAULT;
+    CV_PROP_RW Scalar borderValue = Scalar();
+    CV_PROP_RW int ddepth = -1;
+    CV_PROP_RW double scale = 1.;
+    CV_PROP_RW double shift = 0.;
+};
+
+CV_EXPORTS_AS(filter2Dp) void filter2D( InputArray src, OutputArray dst, InputArray kernel,
+                                        const Filter2DParams& params=Filter2DParams());
```

Comment:
As previously noted this is just a syntax sugar.

- These entries should be generated by automatic script (like binding generator, but for these named parameters)
- This stuff should not be EXPORTed in terms of C++ (its ABI is unstable)
- Implementation should be straightforward and "inline". No extra logic inside!
- Dedicated header should be used (to be generated by script)

Reply:
* previously noted where and by whom?
* we don't care about ABI starting from OpenCV 4.0.
* this will be the main API for complex functions in OpenCV 5 (optical flow, camera calibration, video writer initialization, etc.). The previous API will become backward-compatibility wrappers. The current solution is temporary.
* in this particular case the new API is extension of the previous API. No automatic script can handle that